corehq/apps/locations/models.py

```diff
@@ -592,6 +592,9 @@ class Meta:
         app_label = 'locations'
         unique_together = ('domain', 'site_code',)
 
+    class MPTTMeta:
+        order_insertion_by = ['name']
```

Comment:
`django-mptt` essentially precomputes the order and saves that to the database.  It's done in python before saving, not in SQL during queries (AFAIK).